src/measure_theory/bochner_integration.lean

```diff
@@ -959,6 +1114,89 @@ begin
+  ennreal.to_real (∫⁻ a, ennreal.of_real $ - min (f a) 0) :=
+let f₁ : α →₁ ℝ := l1.of_fun f hfm hfi in
+have eq₁ : ennreal.to_real (∫⁻ a, ennreal.of_real $ max (f a) 0) = ∥l1.pos_part f₁∥ :=
+begin
+  rw l1.norm_eq_norm_to_fun,
+  congr' 1,
+  apply lintegral_congr_ae,
+  filter_upwards [l1.pos_part_to_fun f₁, l1.to_fun_of_fun f hfm hfi],
+  simp only [mem_set_of_eq],
+  assume a h₁ h₂,
+  rw [h₁, h₂, real.norm_eq_abs, abs_of_nonneg],
+  exact le_max_right _ _
+end,
+have eq₂ : ennreal.to_real (∫⁻ a, ennreal.of_real $ -min (f a) 0) = ∥l1.neg_part f₁∥ :=
+begin
+  rw l1.norm_eq_norm_to_fun,
+  congr' 1,
+  apply lintegral_congr_ae,
+  filter_upwards [l1.neg_part_to_fun_eq_min f₁, l1.to_fun_of_fun f hfm hfi],
+  simp only [mem_set_of_eq],
+  assume a h₁ h₂,
+  rw [h₁, h₂, real.norm_eq_abs, abs_of_nonneg],
+  rw [min_eq_neg_max_neg_neg, _root_.neg_neg, neg_zero],
+  exact le_max_right _ _
+end,
+begin
+  rw [eq₁, eq₂, integral, dif_pos],
+  rotate, { exact ⟨hfm, hfi⟩ },
+  exact l1.integral_eq_norm_pos_part_sub _
+end
+
+lemma integral_eq_lintegral_of_nonneg_ae {f : α → ℝ} (hf : ∀ₘ a, 0 ≤ f a) (hfm : measurable f) :
+  integral f = ennreal.to_real (∫⁻ a, ennreal.of_real $ f a) :=
+begin
+  by_cases hfi : integrable f,
+  { rw integral_eq_lintegral_max_sub_lintegral_min hfm hfi,
+    have h_min : (∫⁻ a, ennreal.of_real (-min (f a) 0)) = 0,
+    { rw lintegral_eq_zero_iff,
+      { filter_upwards [hf],
+        simp only [mem_set_of_eq],
+        assume a h,
+        simp only [min_eq_right h, neg_zero, ennreal.of_real_zero] },
+      { refine measurable_of_real.comp
+          (measurable.comp (measurable_neg measurable_id) $
+            measurable_min hfm measurable_const) } },
+    have h_max : (∫⁻ a, ennreal.of_real (max (f a) 0)) = (∫⁻ a, ennreal.of_real $ f a),
+    { apply lintegral_congr_ae,
+      filter_upwards [hf],
+      simp only [mem_set_of_eq],
+      assume a h,
+      rw max_eq_left h },
+    rw [h_min, h_max, zero_to_real, _root_.sub_zero] },
+  { rw integral_eq_zero_of_non_integrable hfi,
+    rw [integrable_iff_norm, lt_top_iff_ne_top, ne.def, not_not] at hfi,
+    have : (∫⁻ (a : α), ennreal.of_real (f a)) = (∫⁻ a, ennreal.of_real ∥f a∥),
+    { apply lintegral_congr_ae,
+      filter_upwards [hf],
+      simp only [mem_set_of_eq],
+      assume a h,
+      rw [real.norm_eq_abs, abs_of_nonneg h] },
+    rw [this, hfi], refl }
+end
+
+lemma integral_of_nonneg_ae {f : α → ℝ} (hf : ∀ₘ a, 0 ≤ f a) : 0 ≤ integral f :=
+begin
+  by_cases hfm : measurable f,
+  { rw integral_eq_lintegral_of_nonneg_ae hf hfm, exact to_real_nonneg },
+  { rw integral_eq_zero_of_non_measurable hfm }
+end
+
+lemma integral_of_nonpos_ae {f : α → ℝ} (hf : ∀ₘ a, f a ≤ 0) : integral f ≤ 0 :=
+begin
+  have hf : ∀ₘ a, 0 ≤ (-f) a,
+  { filter_upwards [hf], simp only [mem_set_of_eq], assume a h, rwa [pi.neg_apply, neg_nonneg] },
+  have : 0 ≤ integral (-f) := integral_of_nonneg_ae hf,
+  rwa [integral_neg, neg_nonneg] at this,
+end
+
```

Comment:
Could you add a lemma `integral_le_integral_of_le_ae` comparing the integrals of two measurable integrable functions? (Just by applying the previous lemma to `f - g`).